matching_test.go

```diff
@@ -0,0 +1,31 @@
+package backoff
+
+import (
+	"errors"
+	"testing"
+)
+
+var (
+	customError    = errors.New("custom error")
+	differentError = errors.New("different error")
+
+	tests = map[error]error{
+		// sanity check
+		customError:    customError,
+		differentError: differentError,
+		// actual test
+		Permanent(customError):    customError,
+		Permanent(differentError): differentError,
+	}
+)
+
+func Test_ErrorMatching(t *testing.T) {
+	for err, expectedType := range tests {
```

Comment:
This is not type but value or pointer depending on what errors.New is returning. To test it properly it would be good to define type implementing error here.